examples/sign_verify.rs

```diff
@@ -0,0 +1,35 @@
+extern crate bitcoin_hashes;
+extern crate secp256k1;
+
+use bitcoin_hashes::{sha256, Hash};
+use secp256k1::{All, Error, Message, PublicKey, Secp256k1, SecretKey, Signature};
+
+fn verify(secp: &Secp256k1<All>, msg: &[u8], sig: [u8; 64], pubkey: [u8; 33]) -> Result<bool, Error> {
+    let msg = sha256::Hash::hash(msg);
+    let msg = Message::from_slice(&msg)?;
+    let sig = Signature::from_compact(&sig)?;
+    let pubkey = PublicKey::from_slice(&pubkey)?;
+
+    Ok(secp.verify(&msg, &sig, &pubkey).is_ok())
+}
+
+fn sign(secp: &Secp256k1<All>, msg: &[u8], seckey: [u8; 32]) -> Result<Signature, Error> {
+    let msg = sha256::Hash::hash(msg);
+    let msg = Message::from_slice(&msg)?;
+    let seckey = SecretKey::from_slice(&seckey)?;
+    Ok(secp.sign(&msg, &seckey))
+}
+
+fn main() {
+    let secp = Secp256k1::new();
+
+    let seckey = [59, 148, 11, 85, 134, 130, 61, 253, 2, 174, 59, 70, 27, 180, 51, 107, 94, 203, 174, 253, 102, 39, 170, 146, 46, 252, 4, 143, 236, 12, 136, 28];
```

Comment:
I don't feel strongly about this, but might be better to use byte strings like `b"\x01\x02\x03\x04"` or to use the `hex` module of the `hashes` crate.

Reply:
Ah my previous suggestion here was actually to include the key generation in both of the examples, i.e., the main function first generates keys, then signs, then verifies. Then we don't have to deal with hard-coded keys at all, which give a misleading impression to the reader, I think.

Reply:
I wanted to simulate getting the keys from another source (which real actual programs will)
If you prefer I can integrate the randomness into it and serialize+deserialize accordingly

Reply:
I would prefer that but I'm fine with the current version. @apoelstra What do you think?

If we keep the current version, can we just add a comment that says "In a real program, you would fetch the secret key from storage here." or similar, and the same for the public key.